warehouse/query-core/src/test/java/datawave/query/UniqueTest.java

```diff
@@ -182,7 +187,10 @@ protected void runTestQueryWithUniqueness(Set<Set<String>> expected, String quer
         TransformIterator iter = new DatawaveTransformIterator(logic.iterator(), transformer);
         List<Object> eventList = new ArrayList<>();
         while (iter.hasNext()) {
-            eventList.add(iter.next());
+            Object o = iter.next();
+            if (o != null) {
+                eventList.add(o);
+            }
```

Comment:
Does this imply that before the transform iterator could return a null element after 'hasNext' was true? Is this a departure from the way things worked before or is the test being brought into alignment with the way the transform iterator actually works?

Reply:
Bringing it into line.  It is valid for a null to be returned when a timeout happens underneath.  This was to allow the RunningQuery to return a null page.  However the RunningQuery was changed to pull the results in a separate thread allow the timeout to be controlled there instead.  I still want to allow the iterators to return a null document for the purposes of avoiding hanging scans in the tservers.